src/liteclient.cc

```diff
@@ -205,6 +207,117 @@ boost::container::flat_map<int, Uptane::Target> LiteClient::getTargets() {
   return sorted_targets;
 }
 
+// Update a device to the given Target
+data::ResultCode::Numeric LiteClient::update(Uptane::Target& target, bool force_update) {
+  data::ResultCode::Numeric update_result = data::ResultCode::Numeric::kUnknown;
+  std::string download_reason;
+  auto current_target = getCurrent();
+
+  Uptane::Target target_to_update{target};
+
+  UpdateType update_type = UpdateType::kNewTargetUpdate;
+  if (force_update) {
+    update_type = UpdateType::kTargetForcedUpdate;
+  } else if (target.filename() == current_target.filename() || !isTargetValid(target)) {
```

Comment:
no, because 

` update_type = UpdateType::kCurrentTargetSync;
    target_to_update = current_target;`
    
   the given if branch for the case when just sync of the current Target is required, in case of the rollback we will sync currently installed version/Target which is not the one that was pending and became kNone (known and never installed) after rollback.